command/agent/agent_endpoint_test.go

```diff
@@ -1019,3 +1022,69 @@ func TestHTTPAgentRegisterServiceCheck(t *testing.T) {
 		t.Fatalf("bad: %#v", result["memcached_check2"])
 	}
 }
+
+func TestHTTPAgent_Monitor(t *testing.T) {
+	logWriter := logger.NewLogWriter(512)
+	expectedLogs := bytes.Buffer{}
+	logger := io.MultiWriter(os.Stdout, &expectedLogs, logWriter)
+
+	dir, srv := makeHTTPServerWithConfigLog(t, nil, logger)
+	srv.agent.logWriter = logWriter
+	defer os.RemoveAll(dir)
+	defer srv.Shutdown()
+	defer srv.agent.Shutdown()
+
+	// Begin streaming logs from the monitor endpoint
+	req, _ := http.NewRequest("GET", "/v1/agent/monitor?loglevel=debug", nil)
```

Comment:
Would be good to do a little deeper test of the level filter. You could just make sure you get an error for an invalid level which proves it gets plumbed down.